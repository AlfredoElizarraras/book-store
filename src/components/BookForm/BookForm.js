import React from 'react';
import { v4 as uuid } from 'uuid';
import BOOK_CATEGORIES from '../../utils/const';

const BookForm = () => (
  <form>
    <label htmlFor="title">
      Book Title:
      <input type="title" id="title" name="title" />
    </label>
    <label htmlFor="categories">
      Choose a car:
      <select id="categories" name="categories">
        {BOOK_CATEGORIES.map(category => (
          <option key={uuid()} value={category}>
            {category}
          </option>
        ))}
      </select>
    </label>
  </form>
);

export default BookForm;
