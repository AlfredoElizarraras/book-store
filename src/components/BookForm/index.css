
@import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@700&display=swap');

.book-form {
  margin-left: 6rem;
  margin-right: 4.4rem;
  margin-bottom: 4rem;
}

.book-form__elements-wrapper {
  margin-top: 1.5rem;
  display: grid;
  grid-template-columns: calc(63% - 1.125rem) calc(22% - 1.125rem) calc(15% - 1.125rem);
  column-gap: 1.125rem;
}

.book-form__title {
  font-family: 'Montserrat', sans-serif;
  font-size: 1.25rem;
  font-weight: bold;
  font-stretch: normal;
  font-style: normal;
  line-height: normal;
  letter-spacing: -0.18px;
  color: #888;
  text-transform: uppercase;
}

input {
  border-radius: 0.25rem;
  padding: 0.813rem 0 0.813rem 1.063rem;
}

.book-form__select-wrapper {
  position: relative;
}

.book-form__select-arrow {
  position: absolute;
  top: 12px;
  right: 15px;
  color: #c4c4c4;
}

.book-form__select {
  height: 100%;
  appearance: none;
  border: 1px solid #c4c4c4;
  margin: 0;
  width: 100%;
  font-family: inherit;
  font-size: inherit;
  border-radius: 0.25rem;
  cursor: pointer;
  line-height: 1.1;
  background-color: #fff;
  color: #c4c4c4;
}
