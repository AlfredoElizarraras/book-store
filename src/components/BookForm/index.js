import { connect } from 'react-redux';
import BookForm from './BookForm';
import { addBook } from '../../actions/index';

export default connect(null, { addBook })(BookForm);
