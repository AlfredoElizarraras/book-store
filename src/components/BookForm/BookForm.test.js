import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import BookForm from './BookForm';

describe('BookForm', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BookForm />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('Has a valid snapshot', () => {
    const component = renderer.create(<BookForm />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
