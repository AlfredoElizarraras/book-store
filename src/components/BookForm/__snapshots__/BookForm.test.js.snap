// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookForm Has a valid snapshot 1`] = `
<form>
  <label
    htmlFor="title"
  >
    Book Title:
    <input
      id="title"
      name="title"
      type="title"
    />
  </label>
  <label
    htmlFor="categories"
  >
    Choose a car:
    <select
      id="categories"
      name="categories"
    >
      <option
        value="Action"
      >
        Action
      </option>
      <option
        value="Biography"
      >
        Biography
      </option>
      <option
        value="History"
      >
        History
      </option>
      <option
        value="Horror"
      >
        Horror
      </option>
      <option
        value="Kids"
      >
        Kids
      </option>
      <option
        value="Learning"
      >
        Learning
      </option>
      <option
        value="Sci-Fi"
      >
        Sci-Fi
      </option>
    </select>
  </label>
</form>
`;
